### PR TITLE
Start Ride paywall button

### PR DESCRIPTION
**Pull request by guytepper**
* changed start ride to open paywall button
* added `useFontFamily` hook for using the `System` font family in English
* added hebrew localization to settings paywall button

<div style="display: flex;">
Simulator Screenshot - iPhone 14 Pro - 2023-05-27 at 14 13 08

Simulator Screenshot - iPhone 14 Pro - 2023-05-27 at 14 12 24
</div>

### CODE REVIEW

File: app/screens/settings/components/better-rail-pro-button.tsx [hunk 1]
```diff
@@ -2,21 +2,24 @@ import { DynamicColorIOS, TouchableOpacity } from "react-native"
...
+import { useFontFamily } from "../../../hooks/use-font-family"
 
 const colors = {
   start: DynamicColorIOS({ light: "#0017E4", dark: "#5E17EB" }),
   end: DynamicColorIOS({ light: "#3793FF", dark: "#9432C2" }),
 }
 
 export function BetterRailProButton({ onPress }) {
+  const { fontFamily, isHeebo } = useFontFamily()
+  console.log(fontFamily)
```
Comment (planecore) [inline app/screens/settings/components/better-rail-pro-button.tsx:14]: no `console.log` in code

File: app/screens/route-details/components/start-ride-button.tsx [hunk 2]
```diff
@@ -64,37 +67,54 @@ export const StartRideButton = observer(function StartRideButton(props: StartRid
...
-              message = translate("ride.rideInPastAlert")
-            } else if (isRouteInFuture) {
-              message = translate("ride.rideInFutureAlert")
-            }
-
-            Alert.alert(message)
+      {!purchases.isPro ? (
+        <Button
+          style={{ backgroundColor: PlatformColor("systemGreen"), width: 148 }}
+          title={translate("ride.liveRide")}
```
Comment (planecore) [inline app/screens/route-details/components/start-ride-button.tsx:73]: Why do we need a different title for the unlock button?

File: app/i18n/he.json [hunk 3]
```diff
@@ -172,14 +174,14 @@
     "month": "חודש",
     "monthlyPrice": "%{price}/חודש",
     "yearlyPrice": "%{price}/שנה",
-    "liveActivity": "עדכונים בזמן אמת",
+    "liveActivity": "בטר רייל לייב",
```
Comment (planecore) [inline app/i18n/he.json:177]: Let's stop with the Hebresh, I promise you Avshalom Kor won't judge 😇

**Review by planecore — CHANGES_REQUESTED** [review]

File: app/screens/paywall/paywall-screen.tsx [hunk 4]
```diff
@@ -95,6 +100,18 @@ export function PaywallScreen({ navigation, route }: PaywallScreenProps) {
     })
   }, [])
 
+  const onPurchase = async () => {
+    try {
+      setPurchaseInProgress(true)
+      await purchases.purchaseOffering(subscriptionType)
+    } catch (error) {
+      // TODO: Add crashlytics report here
+      Burnt.alert({ title: "Something went wrong", preset: "error" })
```
Comment (guytepper) [inline app/screens/paywall/paywall-screen.tsx:109]: @planecore what message do you think is suitable here? what are the failure scenarios?

File: app/i18n/he.json (hunk 3 above)
Comment (guytepper) [reply app/i18n/he.json:176]: doesn't look good in english imo
Simulator Screenshot - iPhone 12 mini - 2023-05-28 at 09 32 16

Comment (planecore) [reply app/i18n/he.json:176]: It looks good IMO!!

File: app/models/purchases/purchases.ts [hunk 5]
```diff
@@ -64,6 +66,15 @@ export const PurchasesModel = types
...
       })
     },
+    async isBetaTester() {
+      return DeviceInfo.getInstallerPackageName().then((value) => {
+        if (value == "TestFlight") {
+          return true
+        }
+
+        return false
+      })
```
Comment (planecore) [inline app/models/purchases/purchases.ts:76]: `return value == "TestFlight"`

File: app/screens/paywall/paywall-screen.tsx (hunk 4 above)
Comment (planecore) [reply app/screens/paywall/paywall-screen.tsx:109]: Maybe "Sorry... Please try again later"

**Review by planecore — CHANGES_REQUESTED** [review]

File: app/screens/paywall/paywall-screen.tsx (hunk 4 above)
Comment (planecore) [reply app/screens/paywall/paywall-screen.tsx:109]: Maybe "Sorry... Please try again later"

**Review by planecore — APPROVED** [review]

**Pull request merged**